test/testdata/resolver/t_struct_subclass.rb

```diff
@@ -0,0 +1,7 @@
+# typed: true
+
+class IS < T::InexactStruct; end
+class Yep < IS; end
+
+class S < T::Struct; end
+class Nope < S; end # error: Subclassing `S` is not allowed
```

Comment:
WDYT aboug adding more tests to demonstrate related cases?
e.g. adding `class Fine < Nope; end;`

Reply:
given this definition for `Nope`, why would `class Fine < Nope; end` be allowed? the definition for `Nope` already issues an error, so I don't see how the definition for `Fine` can be fine.